plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -176,7 +185,12 @@ public PluginCompatReport testPlugins()
 
         // Scan bundled plugins
         // If there is any bundled plugin, only these plugins will be taken under the consideration for the PCT run
-        UpdateSite.Data data = config.getWar() == null ? extractUpdateCenterData(pluginGroupIds) : scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi");
+        UpdateSite.Data data = null;
+        if (config.getBom() != null) {
+            data = scanBom(pluginGroupIds, "([^/.]+)[.][hj]pi");
+        } else {
+            data = config.getWar() == null ? extractUpdateCenterData(pluginGroupIds) : scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi");
+        }
```

Comment:
Interesting but will _not_ fix the problem in #228, since we would still be contacting the UC.

This PR _could_ solve the problem in a different way, by extracting `groupId`s from the BOM including for detached plugins. Perhaps it is already effectively doing that, or we can assume that the BOM manages versions for all detached plugins of interest, in which case this PR could simply be extended to skip `extractUpdateCenterData` in case `-bom` is passed.

Reply:
Thanks for your comments @jglick in this PR we are assuming that BOM manages versions for all detached plugins too, anyway I added [here](https://github.com/jenkinsci/plugin-compat-tester/pull/202/commits/ae8e79232070ee19c41db04381e9209d07e43060#diff-75e1583cf2872b928bd6bfe8c3b8f595R197) the skip of `extractUpdateCenterData` in case bom file is passed.